### PR TITLE
Undeprecate isvalidfilename

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Remove deprecated `isValidFilename`.

### CODE REVIEW

**Comment by juancarlospaco** [comment]
Any news with this one, is there any critical blocker ?. 🙂

**Comment by Araq** [comment]
Instead undeprecate it. It does what its documentation says? What's the issue again, that people use things without reading the docs? What kind of criterion is that?

File: tests/stdlib/tos.nim [hunk 1]
```diff
@@ -664,32 +664,6 @@ block: # normalizePathEnd
     doAssert r"E:/".normalizePathEnd(trailingSep = true) == r"E:\"
     doAssert "/".normalizePathEnd == r"\"
 
-block: # isValidFilename
```
Comment (Araq) [inline tests/stdlib/tos.nim:675]: bring these tests back too.

Comment (juancarlospaco) [reply tests/stdlib/tos.nim:675]: Yes, I want to, the CI was always timing out after ~100 minutes, I think Gitlab was down?.

**Comment by Varriount** [comment]
@Araq The problem is that a "valid" filename depends on both the operating system and file system being used - it's a fundamentally flawed approach. It's better to just... attempt to create a file with a given filename.

See my comment [here](https://github.com/nim-lang/Nim/pull/19004#issuecomment-945062777), from previous discussions regarding this function.

**Comment by Araq** [comment]
> The problem is that a "valid" filename depends on both the operating system and file system being used - it's a fundamentally flawed approach.

I remember the discussion and your point is valid but it's useful enough in practice.

**Pull request merged**